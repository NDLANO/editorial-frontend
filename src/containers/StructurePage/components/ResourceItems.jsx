/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import { ResourceShape } from '../../../shapes';
import Resource from './Resource';
import { deleteTopicResource } from '../../../modules/taxonomy';
import WarningModal from '../../../components/WarningModal';
import { classes } from './ResourceGroup';

class ResourceItems extends React.PureComponent {
  constructor() {
    super();
    this.state = {};
    this.onDelete = this.onDelete.bind(this);
    this.toggleDelete = this.toggleDelete.bind(this);
  }

  async onDelete(id) {
    try {
      this.setState({ deleteId: '' });
      await deleteTopicResource(id);
      this.props.refreshResources();
    } catch (error) {
      console.log(error);
    }
  }

  toggleDelete(id) {
    this.setState({ deleteId: id });
  }

  render() {
    const { contentType, resources, t } = this.props;
    return (
      <ul {...classes('list')}>
        <li {...classes('item')}>
          {resources.map(resource => (
            <Resource
              key={resource.id}
              contentType={contentType}
              name={resource.name}
              onDelete={() => this.toggleDelete(resource.connectionId)}
            />
          ))}
        </li>
        {this.state.deleteId && (
          <WarningModal
            confirmDelete
            text={t('taxonomy.resource.confirmDelete')}
            onContinue={() => this.onDelete(this.state.deleteId)}
            onCancel={() => this.toggleDelete('')}
          />
        )}
      </ul>
    );
  }
}

ResourceItems.propTypes = {
  contentType: PropTypes.string.isRequired,
  resources: PropTypes.arrayOf(ResourceShape),
  classes: PropTypes.func,
  refreshResources: PropTypes.func,
};

export default injectT(ResourceItems);
