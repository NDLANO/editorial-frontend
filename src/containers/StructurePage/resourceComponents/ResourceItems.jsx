/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { ResourceShape } from '../../../shapes';
import Resource from './Resource';
import { deleteTopicResource, updateTopicResource } from '../../../modules/taxonomy';
import { updateRelevanceId } from '../../../util/taxonomyHelpers';
import handleError from '../../../util/handleError';
import MakeDndList from '../../../components/MakeDndList';
import AlertModal from '../../../components/AlertModal';
import { classes } from './ResourceGroup';
import Spinner from '../../../components/Spinner';
import { StructureShape } from '../../../shapes';

const StyledResourceItems = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const StyledErrorMessage = styled.div`
  text-align: center;
`;

class ResourceItems extends React.PureComponent {
  constructor() {
    super();
    this.state = {};
    this.onDelete = this.onDelete.bind(this);
    this.toggleDelete = this.toggleDelete.bind(this);
    this.onDragEnd = this.onDragEnd.bind(this);
  }

  async onDelete(deleteId) {
    try {
      this.setState({ deleteId: '', error: '' });
      await deleteTopicResource(deleteId);
      this.props.refreshResources();
    } catch (e) {
      handleError(e);
      this.setState({
        error: `${this.props.t('taxonomy.errorMessage')}: ${e.message}`,
      });
    }
  }

  async onDragEnd({ destination, source }) {
    if (!destination) {
      return;
    }
    try {
      const { resources, refreshResources } = this.props;
      const { connectionId, primary, relevanceId, rank: currentRank } = resources[source.index];
      const { rank } = resources[destination.index];
      if (currentRank === rank) {
        return;
      }

      this.setState({ loading: true });
      await updateTopicResource(connectionId, {
        primary,
        rank: currentRank > rank ? rank : rank + 1,
        relevanceId,
      });
      await refreshResources();
      this.setState({ loading: false });
    } catch (e) {
      handleError(e.message);
    }
  }

  toggleDelete(deleteId) {
    this.setState({ deleteId });
  }

  render() {
    const { resources, t, currentSubject, structure, locale } = this.props;

    const { deleteId, error, loading } = this.state;

    if (loading) {
      return <Spinner />;
    }
    return (
      <StyledResourceItems {...classes('list')}>
        <MakeDndList onDragEnd={this.onDragEnd} dragHandle>
          {resources.map(resource => (
            <Resource
              resource={resource}
              key={resource.id}
              id={resource.id}
              currentSubject={currentSubject}
              structure={structure}
              onDelete={this.toggleDelete}
              locale={locale}
              updateRelevanceId={updateRelevanceId}
              {...resource}
            />
          ))}
        </MakeDndList>
        {error && (
          <StyledErrorMessage data-testid="inlineEditErrorMessage" {...classes('errorMessage')}>
            {error}
          </StyledErrorMessage>
        )}
        <AlertModal
          show={!!deleteId}
          text={t('taxonomy.resource.confirmDelete')}
          actions={[
            {
              text: t('form.abort'),
              onClick: () => this.toggleDelete(''),
            },
            {
              text: t('alertModal.delete'),
              onClick: () => this.onDelete(deleteId),
            },
          ]}
          onCancel={() => this.toggleDelete('')}
        />
      </StyledResourceItems>
    );
  }
}

ResourceItems.propTypes = {
  resources: PropTypes.arrayOf(ResourceShape),
  classes: PropTypes.func,
  refreshResources: PropTypes.func.isRequired,
  currentSubject: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
  }),
  currentTopic: PropTypes.shape({}),
  structure: PropTypes.arrayOf(StructureShape),
  locale: PropTypes.string,
};

export default withTranslation()(ResourceItems);
