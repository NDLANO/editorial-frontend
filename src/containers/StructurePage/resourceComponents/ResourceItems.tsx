/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { useQueryClient } from '@tanstack/react-query';
import sortBy from 'lodash/sortBy';
import styled from '@emotion/styled';
import { NodeChild } from '@ndla/types-taxonomy';
import { spacing } from '@ndla/core';
import { DragVertical } from '@ndla/icons/editor';
import { DragEndEvent } from '@dnd-kit/core';
import Resource from './Resource';
import handleError from '../../../util/handleError';
import {
  useDeleteResourceForNodeMutation,
  usePutResourceForNodeMutation,
} from '../../../modules/nodes/nodeMutations';
import AlertModal from '../../../components/AlertModal';
import { useTaxonomyVersion } from '../../StructureVersion/TaxonomyVersionProvider';
import {
  NodeResourceMeta,
  resourcesWithNodeConnectionQueryKey,
} from '../../../modules/nodes/nodeQueries';
import { ResourceWithNodeConnectionAndMeta } from './StructureResources';
import { Auth0UserData, Dictionary } from '../../../interfaces';
import DndList from '../../../components/DndList';
import { DragHandle } from '../../../components/DraggableItem';

const StyledResourceItems = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;

const StyledDragHandle = styled(DragHandle)`
  margin-right: ${spacing.xsmall};
`;

const StyledErrorMessage = styled.div`
  text-align: center;
  color: #fe5f55;
`;

interface Props {
  resources: ResourceWithNodeConnectionAndMeta[];
  currentNodeId: string;
  contentMeta: Dictionary<NodeResourceMeta>;
  contentMetaLoading: boolean;
  users?: Dictionary<Auth0UserData>;
}

const isError = (error: unknown): error is Error => (error as Error).message !== undefined;

const ResourceItems = ({
  resources,
  currentNodeId,
  contentMeta,
  contentMetaLoading,
  users,
}: Props) => {
  const { t, i18n } = useTranslation();
  const [deleteId, setDeleteId] = useState<string>('');
  const { taxonomyVersion } = useTaxonomyVersion();

  const qc = useQueryClient();
  const compKey = resourcesWithNodeConnectionQueryKey({
    id: currentNodeId,
    language: i18n.language,
    taxonomyVersion,
  });
  const deleteNodeResource = useDeleteResourceForNodeMutation({
    onMutate: async ({ id }) => {
      await qc.cancelQueries(compKey);
      const prevData = qc.getQueryData<NodeChild[]>(compKey) ?? [];
      const withoutDeleted = prevData.filter((res) => res.connectionId !== id);
      qc.setQueryData<NodeChild[]>(compKey, withoutDeleted);
      return prevData;
    },
  });

  const onUpdateRank = async (id: string, newRank: number) => {
    await qc.cancelQueries(compKey);
    const prevData = qc.getQueryData<NodeChild[]>(compKey) ?? [];
    const updated = prevData.map((r) => {
      if (r.connectionId === id) {
        return { ...r, rank: newRank };
      } else if (r.rank < newRank) {
        return r;
      } else return { ...r, rank: r.rank + 1 };
    });
    qc.setQueryData<NodeChild[]>(compKey, sortBy(updated, ['rank']));
    return resources;
  };

  const { mutateAsync: updateNodeResource } = usePutResourceForNodeMutation({
    onMutate: ({ id, body }) => onUpdateRank(id, body.rank as number),
    onError: (e) => handleError(e),
    onSuccess: () => qc.invalidateQueries(compKey),
  });

  const onDelete = async (deleteId: string) => {
    setDeleteId('');
    await deleteNodeResource.mutateAsync(
      { id: deleteId, taxonomyVersion },
      { onSuccess: () => qc.invalidateQueries(compKey) },
    );
  };

  const onDragEnd = async ({ active, over }: DragEndEvent) => {
    const [source, dest] = [
      resources[active.data.current?.index],
      resources[over?.data.current?.index],
    ];
    if (!dest || !source || source.rank === dest.rank) return;

    await updateNodeResource({
      id: source.connectionId,
      body: {
        primary: source.isPrimary,
        rank: source.rank > dest.rank ? dest.rank : dest.rank + 1,
        relevanceId: source.relevanceId,
      },
      taxonomyVersion,
    });
  };

  const toggleDelete = (newDeleteId: string) => {
    setDeleteId(newDeleteId);
  };

  return (
    <StyledResourceItems>
      <DndList
        items={resources}
        disabled={resources.length < 2}
        onDragEnd={onDragEnd}
        dragHandle={
          <StyledDragHandle aria-label={t('dragAndDrop.handle')}>
            <DragVertical />
          </StyledDragHandle>
        }
        renderItem={(resource) => (
          <Resource
            responsible={
              users?.[contentMeta[resource.contentUri ?? '']?.responsible?.responsibleId ?? '']
                ?.name
            }
            currentNodeId={currentNodeId}
            resource={{
              ...resource,
              contentMeta: resource.contentUri ? contentMeta[resource.contentUri] : undefined,
            }}
            key={resource.id}
            onDelete={toggleDelete}
            contentMetaLoading={contentMetaLoading}
          />
        )}
      />
      {deleteNodeResource.error && isError(deleteNodeResource.error) && (
        <StyledErrorMessage data-testid="inlineEditErrorMessage">
          {`${t('taxonomy.errorMessage')}: ${deleteNodeResource.error.message}`}
        </StyledErrorMessage>
      )}
      <AlertModal
        title={t('taxonomy.deleteResource')}
        label={t('taxonomy.deleteResource')}
        show={!!deleteId}
        text={t('taxonomy.resource.confirmDelete')}
        actions={[
          {
            text: t('form.abort'),
            onClick: () => toggleDelete(''),
          },
          {
            text: t('alertModal.delete'),
            onClick: () => onDelete(deleteId!),
          },
        ]}
        onCancel={() => toggleDelete('')}
      />
    </StyledResourceItems>
  );
};

export default ResourceItems;
