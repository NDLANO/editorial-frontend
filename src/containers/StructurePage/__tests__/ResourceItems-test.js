/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { MemoryRouter } from 'react-router-dom';
import ResourceItems from '../resourceComponents/ResourceItems';
import {
  supplementaryResourcesMock,
  coreResourcesMock,
} from '../../../util/__tests__/taxonomyMocks';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';

afterEach(cleanup);

const wrapper = () =>
  render(
    <MemoryRouter>
      <IntlWrapper>
        <ResourceItems
          resources={[...supplementaryResourcesMock, ...coreResourcesMock]}
          contentType="topic-article"
          activeFilter="filter"
          locale="nb"
          refreshResources={() => {}}
        />
      </IntlWrapper>
    </MemoryRouter>,
  );

test('matches snapshot', () => {
  const { container } = wrapper();
  expect(container.firstChild).toMatchSnapshot();
});
