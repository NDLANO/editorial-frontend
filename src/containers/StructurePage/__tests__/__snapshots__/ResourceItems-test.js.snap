// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can toggle relevance when one filter is chosen 1`] = `
<ul
  class="c-topic-resource__list"
>
  <div
    class="c-dnd-list__drop-zone"
  >
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Kontrast og balanse i grafisk design
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:159742"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Hvorfor bruker vi mediene?
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:166022"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Medieundersøkelser og mediebruk
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:167859"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Mediebruk og statistikk
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:169735"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Identitet og mediebruk
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:178253"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
    <div
      aria-roledescription="Draggable item. Press space bar to lift"
      class="c-dnd-list__drag-item"
      data-react-beautiful-dnd-drag-handle="0"
      data-react-beautiful-dnd-draggable="0"
      draggable="false"
      tabindex="0"
    >
      <li
        class="c-topic-resource__item"
      >
        <div
          class="c-topic-resource__text o-flag o-flag--top"
        >
          <div
            class="c-topic-resource__icon o-flag__img"
          >
            <div
              class="c-content-type-badge c-content-type-badge--subject c-content-type-badge--small c-content-type-badge--background"
            >
              <svg
                class="c-icon"
                data-license="Apache License 2.0"
                data-source="Material Design"
                fill="currentColor"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="vertical-align: middle;"
                viewBox="0 0 24 24"
                width="1em"
              >
                <title>
                  Subject
                </title>
                <g>
                  <title>
                    Emne
                  </title>
                  <defs />
                  <g
                    fill="none"
                    fill-rule="evenodd"
                    id="Page-1"
                    stroke="none"
                    stroke-width="1"
                  >
                    <g
                      fill="currentColor"
                      id="emne"
                    >
                      <path
                        d="M18.3433438,19.3913043 L18.3433438,3.73913043 C9.97639325,3.73913043 5.79291797,3.73913043 5.79291797,3.73913043 C5.79291797,3.73913043 5.79291797,8.95652174 5.79291797,19.3913043 L11.9992148,16.6881419 L18.3433438,19.3913043 Z M17.7457045,2 C18.3688462,2 18.9003414,2.22222 19.3402062,2.66666667 C19.7800709,3.11111333 20,3.63555253 20,4.24 L20,22 L12,18.6666667 L4,22 L4,4.24 C4,3.63555253 4.21992907,3.11111333 4.65979381,2.66666667 C5.09965856,2.22222 5.63115381,2 6.25429553,2 L17.7457045,2 Z"
                        id="bookmark_border---material"
                      />
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
          <div
            class="c-topic-resource__body o-flag__body"
          >
            <h1
              class="c-topic-resource__title"
            >
              Mediepåvirkning
            </h1>
          </div>
          <label
            class="c-toggleSwitch c-toggleSwitch--large"
          >
            <input
              data-testid="toggleRelevance-urn:resource:1:183249"
              type="checkbox"
            />
            <span
              class="c-toggleSwitch__slider c-toggleSwitch__slider--large"
            />
          </label>
          <button
            class="c-button c-button--stripped"
            type="button"
          >
            <svg
              class="c-icon c-topic-resource__deleteIcon"
              data-license="Apache License 2.0"
              data-source="Material Design"
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 24 24"
              width="1em"
            >
              <title>
                RemoveCircle
              </title>
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M7 11v2h10v-2H7zm5-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </g>
            </svg>
          </button>
        </div>
      </li>
    </div>
  </div>
</ul>
`;
